py/pysparkling/context.py

```diff
@@ -132,24 +132,15 @@ def getOrCreate(spark, conf=None, **kwargs):
         print(h2o_context)
 
         # Stop h2o when running standalone pysparkling scripts, only in client deploy mode
-        #, so the user does not explicitly close h2o.
+        #, so the user does not need explicitly close h2o.
         # In driver mode the application would call exit which is handled by Spark AM as failure
         deploy_mode = spark_session.sparkContext._conf.get("spark.submit.deployMode")
         if deploy_mode != "cluster":
-            atexit.register(lambda: h2o_context.stop_with_jvm())
+            atexit.register(lambda: h2o.cluster().shutdown())
```

Comment:
Let's make it more robust - i would not expect that the method will always have time to finish and as a result it will show bunch of random exceptions. 

My proposal is to make the best effort to shutdown in quite mode.

Reply:
I think that this is solution is already quite in all the cases. `h2o.cluster.shutdown()` tries to close the cluster using the Rest API. I tried it in several environments and tests and it closed nicely with no exceptions.

This is the code handling the closing the cluster
```if self._session_id:
            try:
                # If the server gone bad, we don't want to wait forever...
                if self._timeout is None: self._timeout = 1
                self.request("DELETE /4/sessions/%s" % self._session_id)
                self._print("H2O session %s closed." % self._session_id)
            except Exception:
                pass
            self._session_id = None
        self._stage = -1
```

So I think we're fine, what do you think ?

Reply:
btw code ^^ will not shutdown cluster, only delete active objects.

Reply:
ups, sorry. I didn't share the full code. We use the `shutdown` method
```  def shutdown(self, prompt=False):
        """
        Shut down the server.
        This method checks if the H2O cluster is still running, and if it does shuts it down (via a REST API call).
        :param prompt: A logical value indicating whether to prompt the user before shutting down the H2O server.
        """
        if not self.is_running(): return
        assert_is_type(prompt, bool)
        if prompt:
            question = "Are you sure you want to shutdown the H2O instance running at %s (Y/N)? " \
                       % h2o.connection().base_url
            response = input(question)  # works in Py2 & Py3 because redefined in h2o.utils.compatibility module
        else:
            response = "Y"
        if response.lower() in {"y", "yes"}:
            h2o.api("POST /3/Shutdown")
            h2o.connection().close()
```

where the code above is the content of `h2o.connection.close()`

Reply:
but this can still throw an exception, no?

Reply:
hm, only exception could be printed because of `h2o.api("POST /3/Shutdown)`, `h2o.connection().close()` is wrapped in try catch.

So what I can do is to wrap the lambda into `atexit.register(lambda: stop())`
```def stop():
  try: 
      h2o.cluster().shutdown()
  except:
     #ignore everything
```

Reply:
The possible exception is ignored in the latest commit. What do you think about it know pls?